packages/core/src/waitFor.ts

```diff
@@ -44,10 +44,13 @@ export function waitFor<TActorRef extends ActorRef<any, any>>(
   };
   return new Promise((res, rej) => {
     let done = false;
-    const handle = setTimeout(() => {
-      sub.unsubscribe();
-      rej(new Error(`Timeout of ${resolvedOptions.timeout} ms exceeded`));
-    }, resolvedOptions.timeout);
+    const handle =
+      resolvedOptions.timeout === Infinity
```

Comment:
do you perhaps have a link that would mention the valid max value for this? It seems that `Number.MAX_SAFE_INTEGER` also calls this immediately

Reply:
I read that it's a 32bit unsigned int so assumed that the max value was un-specced.

Reply:
https://developer.mozilla.org/en-US/docs/web/api/settimeout#maximum_delay_value

Reply:
I wonder if, in the next major version, the `Infinity` shouldn't be the default. cc @davidkpiano